modules/gateway/peers.go

```diff
@@ -2,10 +2,12 @@ package gateway
 
 import (
 	"errors"
+	"math/rand"
```

Comment:
don't use math/rand in a cryptosystem

it should be fine abusing math/rand still requires a lot of knowledge but it's not as good as crypto/rand.